sdk/modelsrepository/Azure.Iot.ModelsRepository/src/DtmiConventions.cs

```diff
@@ -43,24 +43,20 @@ public static string DtmiToQualifiedPath(string dtmi, string basePath, bool from
                 throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, StandardStrings.InvalidDtmiFormat, dtmi));
             }
 
-            // Normalize directory seperators
-            basePath = basePath.Replace("\\", "/");
-
-            if (!basePath.EndsWith("/", StringComparison.InvariantCultureIgnoreCase))
+            if (expanded)
             {
-                basePath += "/";
+                dtmiPath = dtmiPath.Replace(
+                    ModelsRepositoryConstants.JsonFileExtension,
+                    ModelsRepositoryConstants.ExpandedJsonFileExtension);
             }
 
-            string fullyQualifiedPath = $"{basePath}{dtmiPath}";
-
-            if (fromExpanded)
+            UriBuilder repositoryUriBuilder = new UriBuilder(repositoryUri);
+            if (!repositoryUriBuilder.Path.EndsWith("/", StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Probably should add a code comment as to why we want the URI to end in a slash.

Reply:
Added in new PR https://github.com/Azure/azure-sdk-for-net/pull/19429